torch/csrc/jit/codegen/cuda/lower_alias_memory.cpp

```diff
@@ -71,151 +62,145 @@ class SymbolicSizePrinter final : private OptOutConstDispatch {
         }
       }
     }
 
     // Assume the first argument contains the primary variable
     // Follow path along point-wise operations
-    if (!expr_inputs.empty()) {
-      auto first_input_argument_tv = expr_inputs.front()->getOrigin();
-      if (first_input_argument_tv != nullptr) {
-        return findCompatibleInputAllocate(
-            output_size_str, first_input_argument_tv);
+    if (first_tv_input != nullptr) {
+      if (const auto def = first_tv_input->definition()) {
+        return findCompatibleInputAllocate(output_size_str, def);
       }
     }
+
     return nullptr;
   }
 
-  void handle(Expr* expr) final {
-    size_t expr_index = map_expr_to_pos_.size();
+  void handle(kir::Expr* expr) {
+    const size_t expr_index = map_expr_to_pos_.size();
     map_expr_to_pos_[expr] = expr_index;
 
     if (ir_utils::isTVOp(expr)) {
-      const auto output = expr->output(0)->as<TensorView>();
-      map_tv_to_origin_expr_[output] = expr;
-
-      bool has_allocation = map_tv_to_allocations_.find(output->name()) !=
-          map_tv_to_allocations_.end();
+      const auto output_tv = expr->outputs()[0]->as<kir::TensorView>();
 
-      if (has_allocation) {
-        bool smem_valid = output->getMemoryType() == MemoryType::Shared;
+      const auto alloc_it = map_tv_to_allocations_.find(output_tv->name());
+      if (alloc_it != map_tv_to_allocations_.end()) {
+        const bool smem_valid = (output_tv->memoryType() == MemoryType::Shared);
 
         bool local_valid = false;
-        if (output->getMemoryType() == MemoryType::Local) {
-          auto allocation = map_tv_to_allocations_[output->name()];
-          auto inferred_register_size =
-              eval_evaluator_.inferValue(allocation->size());
-          if (inferred_register_size.has_value()) {
-            local_valid =
-                inferred_register_size.value() > register_size_threshold_;
+        if (output_tv->memoryType() == MemoryType::Local) {
+          const auto allocation = alloc_it->second;
+          const auto register_size =
+              expr_evaluator_.evaluate(allocation->size());
+          if (register_size.has_value()) {
+            local_valid = *register_size > kRegisterSizeThreshold;
           }
         }
 
-        // For the output TV to be an alias candidate,
+        // For the outputv TV to be an alias candidate,
         // its allocation size must exceed the threshold
         // OR be in shared memory
         if (smem_valid || local_valid) {
-          candidate_alias_tv_.insert(output);
+          candidate_alias_tv_.insert(output_tv);
         }
       }
 
-      const auto& expr_inputs =
-          ir_utils::filterByType<TensorView>(expr->inputs());
-      for (const auto input : expr_inputs) {
-        map_tv_to_last_usage_[input] = expr_index;
+      for (auto input : expr->inputs()) {
+        if (auto input_tv = dynamic_cast<kir::TensorView*>(input)) {
+          map_tv_to_last_usage_[input_tv] = expr_index;
+        }
```

Comment:
Is there any reason this is preferred over the previous code?

Reply:
This was accidental (I had bigger changes and missed the opportunity to use `filterByType`), thanks for pointing it out.

Reply:
fixed